receiver/signalfxreceiver/receiver_test.go

```diff
@@ -178,8 +179,10 @@ func Test_signalfxeceiver_EndToEnd(t *testing.T) {
 		}
 		return false
 	}, 10*time.Second, 5*time.Millisecond, "failed to wait for the port to be open")
-	defer exp.Shutdown(context.Background())
-	require.NoError(t, exp.ConsumeMetrics(context.Background(), want))
+	defer func() {
+		require.NoError(t, exp.Shutdown(context.Background()))
+	}()
+	defer require.NoError(t, exp.ConsumeMetrics(context.Background(), want))
```

Comment:
Why defer added here? 
```suggestion
	require.NoError(t, exp.ConsumeMetrics(context.Background(), want))
```